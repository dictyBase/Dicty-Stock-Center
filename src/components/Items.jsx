// @flow
import React from "react"
import { Flex, Box } from "rebass"

type ItemArray = {
  id: number,
  name: string
}

type Props = {
  items: Array<ItemArray>
}

const Items = (props: Props) => {
  const { items } = props
  return (
    <div>
      <Flex wrap>
        <Box w={1 / 2}>
          <h5>ID</h5>
        </Box>
        <Box w={1 / 2}>
          <h5>Strain/Plasmid Name</h5>
        </Box>
      </Flex>
      {items.map((item, index) => {
        return (
          <Flex wrap key={index}>
            <Box w={1 / 2}>{item.id}</Box>
            <Box w={1 / 2}>{item.name}</Box>
          </Flex>
        )
      })}
    </div>
  )
}

export default Items
