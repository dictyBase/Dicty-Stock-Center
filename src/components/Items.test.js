import React from 'react'
import renderer from 'react-test-renderer'
import Items from './Items'

test('matching a snapshot of Items', () => {
    const items = []

    const component = renderer.create(
    <Items items={ items } />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
