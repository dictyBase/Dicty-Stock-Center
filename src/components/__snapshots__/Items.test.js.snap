// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Items 1`] = `
<div>
  <div
    className="sc-bxivhb bUZGRr sc-htpNat jtWIOA sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa goNVXm"
      is={null}
    >
      <h5>
        ID
      </h5>
    </div>
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa goNVXm"
      is={null}
    >
      <h5>
        Strain/Plasmid Name
      </h5>
    </div>
  </div>
</div>
`;
